leetcodePython/SlidingWindow/slidingwindow_632.py

```diff
@@ -0,0 +1,28 @@
+from typing import List
+import heapq
+
+
+class Solution:
+    def smallestRange(self, nums: List[List[int]]) -> List[int]:
+        # start our heap with the first element from every single list
+        heap = [(row[0], i, 0) for i, row in enumerate(nums)]
+        heapq.heapify(heap)
+        ans = [0, float("inf")]
+        # maximum within the smallest data in each row
+        biggest = max(row[0] for row in nums)
+
+        while heap:
+            # get current smallest data from priority queue
+            smallest, r, c = heapq.heappop(heap)
+            # if this data have smaller range, update answer
+            if biggest - smallest < ans[1] - ans[0]:
```

Comment:
the smallest of the smallest in each row, and the biggest of the smallest in each row. That forms the 1st `ans`